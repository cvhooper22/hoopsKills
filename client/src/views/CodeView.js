import React, { Fragment, useState, useEffect } from 'react';
import Loading from '../components/Loading';
import axios from 'axios';
import { withRouter } from 'react-router-dom';

// TODO, change this with the prod build
const urlBase = 'https://byu-ticketing-server.herokuapp.com';

function CodeView (props) {
  const [ imageSource, setImageSource ] = useState('');
  const [ error, setError ] = useState({});
  const [ dropdownOpen, setDropdownOpen ] = useState(false);
  const { history, match } = props;
  const id = match.params.id;
  const imgSrc = '/lareen342576.png';
  //https://byu-ticketing-server.herokuapp.com/qrcode/${id}

  // hold off on this until I set the cors on the server
  // useEffect (() => {
  //   axios.request({ url: `${urlBase}/qrcode/${id}` })
  //     .then((res) => {
  //       // console.log('res', res.data)
  //       setImageSource(res.data);
  //     })
  //     .catch( (err) => {
  //       const errorData = {
  //         message: err.message,
  //       }
  //       if (err.response) {
  //         errorData.message = err.response.data;
  //         errorData.status = err.response.status;
  //       }
  //       setError(errorData);
  //     });
  // }, [])

  function onDrowdownClick () {
    setDropdownOpen(!dropdownOpen);
  }

  function onTryAgain () {
    history.push('/');
  }

  function renderDropdown () {
    if (dropdownOpen) {
      return (
        <Fragment>
          <p className="code-view__instructions__nested flex">
            <i className="material-icons">smartphone</i>
            To download on a mobile device, press and hold on the QR code and an option should show up to download the image.
          </p>
          <p className="code-view__instructions__nested flex">
            <i className="material-icons">desktop_windows</i>
            To download on a computer, right-click on the QR code and choose "Save Image As..."
          </p>
        </Fragment>
      );
    }
    return undefined;
  }

  function renderCode () {
    return (
      <Fragment>
        <img className="code-view__qr-code" src={ imageSource } />
        <div className="code-view__instructions mt-large">
          Download the above image to have a digital copy of your ROC pass.
          <button type="button" className="dropdown-trigger flex ai-c" onClick={ onDrowdownClick }>
            { !dropdownOpen ? 
                <i className="material-icons">keyboard_arrow_right</i>
              :
              <i className="material-icons">keyboard_arrow_down</i>
            }
            How do I download?
          </button>
          { renderDropdown() }
        </div>
      </Fragment>
    )
  }

  function renderError () {
    return (
      <div className="code-view__error flex column ai-c">
        <div className="code-vew__error__message">
          { renderErrorMessage () }
        </div>
        <button className="mt-reg" type="button" onClick={ onTryAgain }>Try Again</button>
      </div>
    );
  }

  function renderErrorMessage () {
    if (error.status === 404) {
      return (
        <Fragment>
          Could not find a QR code for the id that you entered.
          <ul>
            <li>Did you enter your Net ID and not your Student ID number?</li>
            <li>Did you remove all dashes from your Student ID number?</li>
            <li>Have you received an email letting you know your digital pass should be ready?</li>
          </ul>
        </Fragment>
      );
    } else if ( error.message) {
      return `There was an error generating your digital ROC Pass: ${error.message}`;
    }
    return 'An error occured generating your digital ROC Pass'
  }

  function renderLoading () {
    return (
      <div className="code-view__loading flex column ai-c">
        <Loading />
        <div className="code-view__loading__message mt-reg">Generating your digital pass...</div>
      </div>
    );
  }

  function renderContent () {
    if (imageSource.length) {
      return renderCode();
    } else if (Object.keys(error).length) {
      return renderError();
    }
    return renderLoading();
  }

  return (
    <div className="code-view flex column ai-c">
      { renderContent() }
    </div>
  );
}

export default withRouter(CodeView);
