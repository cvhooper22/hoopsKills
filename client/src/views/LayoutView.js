import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

function LayoutView (props) {
  const { renderPageContent } = props;

  return (
    <Fragment>
      <div className="container">
        <div className="row">
          <div className="col-md-12">
            { renderPageContent() }
          </div>
        </div>
      </div>
    </Fragment>
  );
}

LayoutView.propTypes = {
  renderPageContent: PropTypes.func,
};

export default LayoutView;
