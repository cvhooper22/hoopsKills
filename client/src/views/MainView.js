import React, { useState } from 'react';
import { withRouter } from 'react-router-dom';

function MainView (props) {
  const imgSrc = './ROClogoclear.png';
  const [ id, setId ] = useState('');
  const { history } = props;

  function onSubmit(evt) {
    if (id.length) {
      console.log('onsubmit', evt);
      console.log('id is', id);
      history.push(`/qrcode/${id}`);
    }
  }
  return (
    <div className="main__container flex column ai-c">
      <img className="main__splash-image" src={ imgSrc } />
      <div className="main__instructions mt-reg">
        Here there be stats
      </div>
      <input className="main__id-input mt-reg" type="text" placeholder="Student Id" value={ id } onChange={ e => setId(e.target.value) } />
      <button className="mt-reg" type="button" onClick={ onSubmit }>Get QR Code</button>
    </div>
  );
}

export default withRouter(MainView);
