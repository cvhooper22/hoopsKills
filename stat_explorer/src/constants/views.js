const LINEUP_VIEW = "Lineups";
const CLUTCH_VIEW = "Clutch";
const KILLS_VIEW = "Kills";
const HOME_VIEW = "Home";
const views = [
    {
      title: HOME_VIEW,
      route: '/'
    },
    {
      title: LINEUP_VIEW,
      route: '/lineups/san-fran-a'
    },
    {
      title: KILLS_VIEW,
      disabled: true,
      route: '/kills'
    },
    {
      title: CLUTCH_VIEW,
      disabled: true,
      route: '/clutch'
    } 
  ];
export { views, HOME_VIEW, LINEUP_VIEW, CLUTCH_VIEW, KILLS_VIEW };