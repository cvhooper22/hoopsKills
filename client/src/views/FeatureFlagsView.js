import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { useTranslation } from 'react-i18next';

import Title from '../components/Title';
import LayoutView from './LayoutView';

const mapStateToProps = (state) => {
  return {
    featureFlags: state.featureFlags.featureFlags,
  };
};

export function ConnectedFeatureFlagsView (props) {
  const { t } = useTranslation();
  const { featureFlags } = props;

  function renderFeatureFlags () {
    const ffString = t('views.featureflagsview.featureflag');
    return Object.keys(featureFlags).map((key) => {
      return <p key={ `${key}` }>{ `${ffString}${key} = ${featureFlags[key]}` }</p>;
    });
  }

  function renderPageContent () {
    return (
      <Fragment>
        <Title
          title={ t('views.featureflagsview.title') }
        ></Title>
        <div className="section">
          <div className="row">
            <div className="col-md-12">
              { renderFeatureFlags() }
            </div>
          </div>
        </div>
      </Fragment>
    );
  }

  return (
    <LayoutView
      renderPageContent={ renderPageContent }
    ></LayoutView>
  );
}

ConnectedFeatureFlagsView.propTypes = {
  date: PropTypes.string,
  featureFlags: PropTypes.object,
};

const FeatureFlagsView = connect(mapStateToProps)(ConnectedFeatureFlagsView);

export default FeatureFlagsView;
