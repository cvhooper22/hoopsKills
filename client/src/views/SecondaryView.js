import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { useTranslation } from 'react-i18next';

import Title from '../components/Title';
import LayoutView from './LayoutView';
import Loading from '../components/Loading';
import ConfigGrid from '../components/ConfigGrid';

const mapStateToProps = (state) => {
  return {
    appReady: state.environment.appReady,
    config: state.config.config,
  };
};

export function ConnectedSecondaryView (props) {
  const { appReady, config } = props;
  const { t } = useTranslation();

  function renderPageContent () {
    if (appReady && config) {
      return (
        <Fragment>
          <Title
            title={ t('views.secondaryview.title') }
          ></Title>
          <ConfigGrid
            data={ config }
          ></ConfigGrid>
        </Fragment>
      );
    } else {
      return <Loading />;
    }
  }

  return (
    <LayoutView
      renderPageContent={ renderPageContent }
    ></LayoutView>
  );
}

ConnectedSecondaryView.propTypes = {
  appReady: PropTypes.bool,
  config: PropTypes.array,
};

const SecondaryView = connect(mapStateToProps)(ConnectedSecondaryView);

export default SecondaryView;
