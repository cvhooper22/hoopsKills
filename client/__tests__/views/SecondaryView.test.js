import React from 'react';
import renderer from 'react-test-renderer';
import { I18nextProvider } from 'react-i18next';

import { ConnectedSecondaryView } from '../../src/views/SecondaryView';
import i18n from '../i18n';

describe('SecondaryView -', () => {
  const config = [
    { id: 'item1', label: 'item one' },
    { id: 'item2', label: 'item two' },
    { id: 'item3', label: 'item three' },
  ];

  it('Should render SecondaryView', () => {
    const view = renderer.create(
      <I18nextProvider i18n={ i18n }>
        <ConnectedSecondaryView
          appReady
          config={ config }
        ></ConnectedSecondaryView>
      </I18nextProvider>
    );

    expect(view).toMatchSnapshot();
  });

});