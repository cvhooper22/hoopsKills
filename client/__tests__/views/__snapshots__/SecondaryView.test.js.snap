// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryView - Should render SecondaryView 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <h1>
        <i
          className="ids-icon__dashboard"
        />
         Test - Secondary view
      </h1>
      <table>
        <tbody>
          <tr>
            <td>
              item one
            </td>
          </tr>
          <tr>
            <td>
              item two
            </td>
          </tr>
          <tr>
            <td>
              item three
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
